File: crates/chidori-js/benchmarks/workloads/array_push_sum.js
// Array growth + indexed read loop.
// Mirrors the `array_push_sum` criterion micro-benchmark, scaled up.
const N = 500_000;
const a = [];
for (let i = 0; i < N; i++) a.push(i);
let s = 0;
for (let i = 0; i < a.length; i++) s += a[i];
console.log("RESULT=" + s);

File: crates/chidori-js/benchmarks/workloads/array_push_sum.py
# Python twin of array_push_sum.js — list growth + indexed read loop.
# Indexed `for i in range(len(a))` (not `for x in a`) to mirror the JS loop.
N = 500_000
a = []
for i in range(N):
    a.append(i)
s = 0
for i in range(len(a)):
    s += a[i]
print("RESULT=" + str(s))
